src/app/shared/components/template/services/template-calc-functions/plh-calc-functions.ts

```diff
@@ -0,0 +1,156 @@
+import { IFunctionHashmap } from "packages/shared/src";
+
+/**
+ * Temporary functions used for plh sheets
+ * Eventually this will be migrated to read from data_list instead (once able to define functions within list)
+ */
+export const PLH_CALC_FUNCTIONS: IFunctionHashmap = {
+  /**
+   * Add a new family to the list of all families
+   * Family members are identified uniquely by string id, avoiding duplication if already exists
+   * @param families Array of all families, e.g. [['Ada','Blaise'],['Charles']]
+   * @param members List of members to add to family, e.g. ['Daniel','Eva']
+   * @returns Combined list [['Ada','Blaise'],['Charles'],['Daniel','Eva']]
+   */
+  plh_add_family: (families: any, ...members: string[]) => {
+    // Handle case where families is passed as a string representation (assumed to be of string[][])
```

Comment:
NB this code is re-used in each function, but because of the way the functions are injected for use in `@calc()` blocks, they must explicitly contain all of their logic, so I haven't extracted this logic into an additional function

Reply:
Makes sense, although would probably be nicer if type-casting were handled in another way (e..g using local/list variables that already are formatted as arrays, but guessing not as easy to do for some values?)

Can we use functions declared in calc? e.g. can `plh_add_family` call another defined `plh_parse_family` or similar?

Reply:
@jfmcquade 
Turns out yes to the common parsing method - simply defining a standalone function can then be used across all other functions. I've made the small refactor (and added a test case) with [3e22f5d](https://github.com/IDEMSInternational/parenting-app-ui/pull/2127/commits/3e22f5d5e0ff8f206bcacfb434a044b168d379b6)

I've also similarly moved the _core_ calc functions to their own file with [0527862](https://github.com/IDEMSInternational/parenting-app-ui/pull/2127/commits/0527862fae711c1d3789f25041a7c57457beba8a)
(yet to add specs, will probably do so with follow-up pr #2130)